File: Python/Binary_Search.py
def BinearSearch(arr, size, element):
	low = 0
	high = size - 1
	while low <= high:
		mid = int((low + high)/2)
		if arr[mid] == element:
			return mid
		if arr[mid] < element:
			low = mid + 1
		else:
			high = mid - 1
	return -1
	
arr = [12, 29, 48, 63, 84]
size = len(arr)
element = 63
pos = BinearSearch(arr, size, element)

if pos == -1:
	print("{} is not found".format(element))
else:
	print("Element found at index {}".format(pos))

File: C Language/Binary_Search.c
#include<stdio.h>

int BinearSearch(int arr[], int size, int element)
{
    int low, mid, high;
    low = 0;
    high = size-1;
    while(low <= high)
    {
        mid = (low + high)/2;
        if(arr[mid] == element)
        {
            return mid;
        }
        if(arr[mid] < element)
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
    }
    return -1;
}

int main()
{
    int arr[] = {23, 27, 34, 85,96};
    int size;
    int element = 27;
    size = sizeof(arr)/sizeof(int);
    int pos = BinearSearch(arr, size, element);

    if(pos == -1)
        printf("%d was not found", element);
    else
        printf("The element %d was found at %d\n",element, pos);
    return 0;
}